test/integration/errors_test.go

```diff
@@ -277,3 +281,44 @@ func TestBadSignatureAlgorithm(t *testing.T) {
 		t.Error("problem document MUST contain acceptable algorithms, got none")
 	}
 }
+
+// TestOrderFinalizeEarly tests that finalizing an order before it is fully
+// authorized results in an orderNotReady error.
+func TestOrderFinalizeEarly(t *testing.T) {
+	t.Parallel()
+
+	client, err := makeClient()
+	if err != nil {
+		t.Fatalf("creating acme client: %s", err)
+	}
+
+	domain := randomDomain(t)
+	idents := []acme.Identifier{{Type: "dns", Value: domain}}
```

Comment:
Because `domain` is not used anywhere else in the test:
```suggestion
	idents := []acme.Identifier{{Type: "dns", Value: randomDomain(t)}}
```